domain/src/test/resources/com/krzykrucz/elesson/currentlesson/StudentUnprepared.feature

```diff
@@ -0,0 +1,47 @@
+Feature: Student reports unprepared to lesson
+
+  Scenario: Should note student unprepared
+    Given Present Ron Weasley from class 1A
+    And Ron Weasley reported unprepared 2 times in a semester
+    And Empty list of unprepared students
+    And Lesson after attendance checked but before topic assigned
+    When Ron Weasley reports unprepared
+    Then Ron Weasley should be noted unprepared for lesson
+#    And Ron Weasley should've been unprepared 3 times in the semester
+
+  Scenario: Should not note not present student unprepared
+    Given Present Ron Weasley from class 1A
+    And Ron Weasley reported unprepared 2 times in a semester
+    And Empty list of unprepared students
+    And Lesson after attendance checked but before topic assigned
+    When Harry Potter reports unprepared
+    Then Harry Potter should not be noted unprepared for lesson
+#    And Ron Weasley should've been unprepared 2 times in the semester
+
+  Scenario: Should not note not present student unprepared
+    Given Present Ron Weasley from class 1A
+    And Ron Weasley reported unprepared 2 times in a semester
+    And Empty list of unprepared students
+    And Lesson after topic assigned
+    When Ron Weasley reports unprepared
+    Then Ron Weasley should not be noted unprepared for lesson
+#    And Ron Weasley should've been unprepared 2 times in the semester
+
+  Scenario: Should not note student unprepared twice
+    Given Present Ron Weasley from class 1A
+    And Ron Weasley reported unprepared 2 times in a semester
+    And Ron Weasley on the list of unprepared students
+    And Lesson after attendance checked but before topic assigned
+    When Ron Weasley reports unprepared
+    Then Ron Weasley should not be noted unprepared for lesson
+#    And Ron Weasley should've been unprepared 2 times in the semester
+
+  Scenario: Should not note student unprepared too many times in the semester
+    Given Present Ron Weasley from class 1A
+    And Ron Weasley reported unprepared 3 times in a semester
+    And Empty list of unprepared students
+    And Lesson after attendance checked but before topic assigned
+    When Ron Weasley reports unprepared
+    Then Ron Weasley should not be noted unprepared for lesson
+#    And Ron Weasley should've been unprepared 3 times in the semester
```

Comment:
a couple of commented assertions, why are they here?

Reply:
I will remove them, this is moved to a read model